app/auth-portal/src/content/tutorial/08-asset_authoring.md

```diff
@@ -0,0 +1,442 @@
+to a `region` asset so we can create an input socket for a Region.
+
+When creating a socket, we need to give that socket an `arity`. The arity is the number of
+connections it can make. An EBS Volume can only be in a single region so the region input socket
+has an arity of `one`. For my use case, I am going to attach the Volume to a single instance at a
+time, so I have made the output socket of volume ID have an arity of `one`.
+
+In System Initiative, we have a loose type system in connecting the assets. We *currently* use the
+name of the sockets to do that, so `Region` as an input socket name matches the `Region` output socket
+name of the region asset.
+
+Now that we have built out the structure of the code, let's update the metadata of the asset and create it:
+
+![create-asset.png](/tutorial-img/08-asset-authoring/create-asset.png)
+
+When we whack the `create asset` button we will get a popup to let us know that we have a new asset
+
+![create-success.png](/tutorial-img/08-asset-authoring/create-success.png)
+
+We have just built our schema definition. We can now go and attach the behaviour to our asset to make
+it do what we need it to.
+
+### Authoring a Code Generation Function
+
+In order to create an EBS volume, we are going to model the data that we want to send to AWS.
+As we will be re-using this data structure, we are going to create a codegen func. Select the asset we just created
+and use the `Attach Function` button and select `New function`. The UI gives us a pop-up and allows us
+to select the kind of function.
+
+I've chosen `Code Generation` and set the name to be `awsEc2EbsVolumeJSON` - the name of this function is important as
+it will be used as part of the qualification and action functions.
+
+![create-codegen.png](/tutorial-img/08-asset-authoring/create-codegen.png)
+
+When we click the button, we get a pre-generated code snippet in the function editor. Remove the pre-generated
+code and replace it with the following:
+
+```js
+async function generateCode(input: Input): Promise<Output> {
+  const object = {
+    AvailabilityZone: input.domain?.availabilityZone,
+    Iops: input.domain?.iops,
+    VolumeType: input.domain?.volumeType,
+    Size: input.domain?.size,
+    Encrypted: false, // We are defaulted to non-encryped right now
+  };
+
+  return {
+    format: "json",
+    code: JSON.stringify(object, null, "\t"),
+  };
+}
+```
+
+Use the `Execute` button to be able to save and update the function attachment.
+
+![codegen-func-attachment.png](/tutorial-img/08-asset-authoring/codegen-func-attachment.png)
+
+*~> Notice, a display name is necessary for the function, so I have made the display name the same as the function name*
+
+We are now able to use this code generation function to be able to qualify and create the resource.
+
+### Authoring a Qualification Function
+
+A qualification function is used to understand if the resource we are attempting to create is going to work as expected.
```

Comment:
A qualification function is used to understand if the resource we are attempting to create is going to work as expected - like a real time test to give us fast feedback on whether the system we propose is configured correctly.